public/app/plugins/datasource/cloudwatch/datasource.js

```diff
@@ -379,7 +389,7 @@ function (angular, _, moment, dateMath, kbn, CloudWatchAnnotationQuery) {
 
         if (dimensionKey) {
           var variable = _.find(templateSrv.variables, function(variable) {
-            return templateSrv.containsVariable(target.dimensions[dimensionKey], variable.name);
+            return self.containsVariable(target.dimensions[dimensionKey], variable.name);
```

Comment:
Short answer, the code already did contain a contains check. I'll see if I have time to look into it later this week. You should feel free to just create a fix such that it will be working on the master again.